corehq/messaging/tasks.py

```diff
@@ -89,15 +89,17 @@ def initiate_messaging_rule_run(domain, rule_id):
     transaction.on_commit(lambda: run_messaging_rule.delay(domain, rule_id))
 
 
+def paginated_case_ids(domain, case_type):
+    for case in paginate_query_across_partitioned_databases(CommCareCaseSQL,
+                                                            Q(domain=domain, type=case_type, deleted=False)):
+        yield case.case_id
```

Comment:
`paginate_query_across_partitioned_databases` should be changed to support a `values` parameter so that we are only requesting the case_id from postgres instead of the entire case (as was the case in the previous code)